Marketplace/gestureRose/gestureRose.js

```diff
@@ -0,0 +1,114 @@
+var ROSE_URL = Script.resolvePath('longStemRose.fbx');
+
+var gestureStarted = false;
+var gestureStartPos = null;
+var _gestureHand = 0;
+
+Script.scriptEnding.connect(function () {
+    shutdown();
+});
+
+controllerMappingName = 'gestureRoseMapping';
+controllerMapping = Controller.newMapping(controllerMappingName);
+controllerMapping.from(Controller.Standard.LeftGrip).to(function (value) {
+    gestureLogic(value,-1);
+});
+controllerMapping.from(Controller.Standard.RightGrip).to(function (value) {
+    gestureLogic(value,1);
+});
+
+function gestureLogic(value,gestureHand){
+    var jointName = "LeftHand";
+    if(gestureHand === 1){
+        jointName = "RightHand";
+    }
+    if (!gestureStarted && (value > GRIP_PULL) && isPalmUpwards(gestureHand)) {
+        // Grip started
+        _gestureHand = gestureHand;
+        gestureStarted = true;
+        gestureStartPos = MyAvatar.getJointPosition(jointName);
+    } else if (gestureStarted && (value < GRIP_RELEASE)) {
+        //  Grip ended, check if it was the right gesture
+        if (isPalmUpwards(gestureHand) && gestureStarted && gestureHand === _gestureHand) {
+            var gestureEndPos = MyAvatar.getJointPosition(jointName);
+            var total = Vec3.subtract(gestureEndPos, gestureStartPos);
+            if (Math.abs(total.x) < X_GESTURE_LIMIT && total.y < Y_GESTURE_DIST && Math.abs(total.z) < Z_GESTURE_LIMIT) {
+                var clientOnly = !(Entities.canRez() || Entities.canRezTmp());
+                //  Correct gesture, make a rose. 
+                Entities.addEntity({
+                    name: "Long Stemmed Rose",
+                    shapeType: "simple-hull",
+                    type: "Model",
+                    modelURL: ROSE_URL,
+                    position: Vec3.sum(gestureEndPos, Vec3.multiplyQbyV(MyAvatar.orientation, ROSE_REZ_OFFSET)),
+                    dimensions: ROSE_SCALE,
+                    dynamic: true,
+                    lifetime: ROSE_LIFETIME,
+                    gravity: ROSE_DEFAULT_GRAVITY,
+                    userData: "{ \"grabbableKey\": { \"grabbable\": true, \"kinematic\": false } }"
+                }, clientOnly);
+            }
+        }
+        gestureStarted = false;
+    }
+}
+
+init();
+
+function isPalmUpwards(gestureHand) {
+    var handRot = MyAvatar.getLeftHandPose()["rotation"];
+    if(gestureHand === 1){
+        handRot = MyAvatar.getRightHandPose()["rotation"];
+    }
+    var returnVal = false;
+    var normal = Vec3.multiplyQbyV(Quat.multiply(MyAvatar.orientation, handRot), {
+        x: 0,
+        y: 0,
+        z: 1
+    });
+    var angle = Vec3.dot(GLOBAL_UP, normal);
+    if (angle > ANGLE_MATCH) {
+        returnVal = true;
+    }
+    return returnVal;
+}
+
+function init() {
+    Controller.enableMapping(controllerMappingName);
+}
+
+
+function shutdown() {
+    try {
+        Controller.disableMapping(controllerMappingName);
+    } catch (e) {
+        // empty
+    }
+}
```

Comment:
Newline

Reply:
I don't understand this one, did you want it as just 
{
}

Reply:
It just needs a new empty line at the end of the file

Reply:
Done